pkg/overlay/server.go

```diff
@@ -74,52 +81,76 @@ func (server *Server) FindStorageNodes(ctx context.Context, req *pb.FindStorageN
 	defer mon.Task()(&ctx)(&err)
 
 	opts := req.GetOpts()
-	maxNodes := req.GetMaxNodes()
-	if maxNodes <= 0 {
-		maxNodes = opts.GetAmount()
+	// todo(nat): change name from MaxNodes to minRequestedNodes or similar
```

Comment:
Let's make this logic part of cache (later renamed to Service), it makes  it easier to test as well.